libbeat/dashboards/kibana_loader.go

```diff
@@ -166,6 +171,35 @@ func (loader KibanaLoader) ImportDashboard(file string) error {
 	return nil
 }
 
+type dashboardObj struct {
+	References []dashboardReference `json:"references"`
+}
+type dashboardReference struct {
+	ID   string `json:"id"`
+	Type string `json:"type"`
+}
+
+func (loader KibanaLoader) importReferences(path string, dashboard []byte) error {
+	var d dashboardObj
+	err := json.Unmarshal(dashboard, &d)
+	if err != nil {
+		return fmt.Errorf("failed to parse dashboard references: %+v", err)
+	}
+
+	base := filepath.Dir(path)
+	for _, ref := range d.References {
+		if ref.Type == "index-pattern" {
+			continue
+		}
+		referencePath := filepath.Join(base, "..", ref.Type, ref.ID+".json")
+		err := loader.ImportDashboard(referencePath)
```

Comment:
Adrian mentioned that Kibana "fails" with status 200 if it cannot import any dashboard because of missing references. Should we check the content of the response in `ImportDashboards` so we detect this kind of issues earlier?

Reply:
Updated.